### PR TITLE
Remove auth null checks in `ExerciseController` and `HistoryController` methods

### LINKED ISSUE #83: Remove auth null checks in `ExerciseController` methods
`Auth` null check is unnecessary as the `@PreAuthorize("authenticated")` annotation should make sure the methods cannot be called by anonymous users.

Remove `auth != null` from all these:

https://github.com/mepox/fitbuddy/blob/5558b9ace16378be9474919128c90e9c4f5d0a7f/src/main/java/com/laszlojanku/fitbuddy/operation/controller/ExerciseController.java#L47

https://github.com/mepox/fitbuddy/blob/5558b9ace16378be9474919128c90e9c4f5d0a7f/src/main/java/com/laszlojanku/fitbuddy/operation/controller/ExerciseController.java#L60

https://github.com/mepox/fitbuddy/blob/5558b9ace16378be9474919128c90e9c4f5d0a7f/src/main/java/com/laszlojanku/fitbuddy/operation/controller/ExerciseController.java#L74

https://github.com/mepox/fitbuddy/blob/5558b9ace16378be9474919128c90e9c4f5d0a7f/src/main/java/com/laszlojanku/fitbuddy/operation/controller/ExerciseController.java#L88

### LINKED ISSUE #84: Remove auth null checks in `HistoryController` methods
`Auth` null check is unnecessary as the @PreAuthorize("authenticated") annotation should make sure the methods cannot be called by anonymous users.

Remove `auth != null` from all these:

https://github.com/mepox/fitbuddy/blob/5558b9ace16378be9474919128c90e9c4f5d0a7f/src/main/java/com/laszlojanku/fitbuddy/operation/controller/HistoryController.java#L48

https://github.com/mepox/fitbuddy/blob/5558b9ace16378be9474919128c90e9c4f5d0a7f/src/main/java/com/laszlojanku/fitbuddy/operation/controller/HistoryController.java#L61

https://github.com/mepox/fitbuddy/blob/5558b9ace16378be9474919128c90e9c4f5d0a7f/src/main/java/com/laszlojanku/fitbuddy/operation/controller/HistoryController.java#L82

https://github.com/mepox/fitbuddy/blob/5558b9ace16378be9474919128c90e9c4f5d0a7f/src/main/java/com/laszlojanku/fitbuddy/operation/controller/HistoryController.java#L96

### PR DESCRIPTION
**Pull request by dmitriydb**
## Close issue(s) 
Close #83 
Close #84

### CODE REVIEW

**Comment by mepox** [comment]
Hi @dmitriydb ! Thanks for your contribution. I see you committed both changes into this PR. This is fine for now as both tasks were small. I will close the other PR #91.

**Comment by dmitriydb** [comment]
> Hi @dmitriydb ! Thanks for your contribution. I see you committed both changes into this PR. 

Hi. Sorry, it was unintentional. 
I merged the first issue branch into my main branch before branching into another issue.

I'm glad it's okay, will be more careful next time

**Comment by mepox** [comment]
> > Hi @dmitriydb ! Thanks for your contribution. I see you committed both changes into this PR.
> 
> @mepox Hi. Sorry, it was unintentional. I merged the first issue branch into my main branch before branching into another issue.
> 
> I'm glad it's okay, will be more careful next time

@dmitriydb No worries 👍 I will submit a review in the next few minutes.

File: src/main/java/com/laszlojanku/fitbuddy/operation/controller/ExerciseController.java [hunk 1]
```diff
@@ -45,48 +45,43 @@ public ExerciseController(ExerciseCrudService exerciseCrudService,
...
 		Authentication auth = SecurityContextHolder.getContext().getAuthentication();
-		if (auth != null) {
 			Integer userId = appUserCrudService.readByName(auth.getName()).getId();
 			if (userId != null) {				
 				exerciseDto.setAppUserId(userId);	
 				exerciseCrudService.create(exerciseDto);
 				logger.info("Creating new exercise: {}", exerciseDto);
 			}			
-		}	
 	}
```
Comment (mepox) [inline src/main/java/com/laszlojanku/fitbuddy/operation/controller/ExerciseController.java:54]: Indentation from Line 48

File: src/main/java/com/laszlojanku/fitbuddy/operation/controller/ExerciseController.java [hunk 2]
```diff
@@ -45,48 +45,43 @@ public ExerciseController(ExerciseCrudService exerciseCrudService,
...
-			AppUserDto appUserDto =  appUserCrudService.readByName(auth.getName());						
+			AppUserDto appUserDto =  appUserCrudService.readByName(auth.getName());
 			if (appUserDto != null && appUserDto.getId() != null) {
 				List<ExerciseDto> dtos = exerciseCrudService.readMany(appUserDto.getId());
 				logger.info("Sending a list of exercises.");
 				return dtos;
 			}
-		}
 		return null;
 	}
```
Comment (mepox) [inline src/main/java/com/laszlojanku/fitbuddy/operation/controller/ExerciseController.java:66]: Indentation

File: src/main/java/com/laszlojanku/fitbuddy/operation/controller/HistoryController.java [hunk 3]
```diff
@@ -59,13 +58,12 @@ public void create(@Valid @RequestBody HistoryDto historyDto) {
 				historyCrudService.create(historyDto);
 				logger.info("Creating new history: {}", historyDto);
 			}
-		}		
 	}
```
Comment (mepox) [inline src/main/java/com/laszlojanku/fitbuddy/operation/controller/HistoryController.java:61]: Indentation

File: src/main/java/com/laszlojanku/fitbuddy/operation/controller/HistoryController.java [hunk 4]
```diff
@@ -47,7 +47,6 @@ public HistoryController(HistoryCrudService historyCrudService, AppUserCrudServi
 	@PostMapping
 	public void create(@Valid @RequestBody HistoryDto historyDto) {
 		Authentication auth = SecurityContextHolder.getContext().getAuthentication();
-		if (auth != null) {		
 			try {
```
Comment (mepox) [inline src/main/java/com/laszlojanku/fitbuddy/operation/controller/HistoryController.java:50]: Indentation

File: src/main/java/com/laszlojanku/fitbuddy/operation/controller/HistoryController.java [hunk 5]
```diff
@@ -47,7 +47,6 @@ public HistoryController(HistoryCrudService historyCrudService, AppUserCrudServi
...
 				LocalDate.parse(historyDto.getCreatedOn());
 			} catch (DateTimeParseException e) {
```
Comment (mepox) [inline src/main/java/com/laszlojanku/fitbuddy/operation/controller/HistoryController.java:52]: Indentation

**Review by mepox — CHANGES_REQUESTED** [review]
@dmitriydb Thanks again for your work. The code in the PR looks good, but as you have removed some if statments completely, you missed to adjust the indentations. 

Indentations are does not matter in Java (comparing to some other languages) for code "runability", but it's good for code readability.

I left some comments above where I found the indentations are not correct. Could you please fix those? Once that done, this PR will be merged. 👍

**Comment by dmitriydb** [comment]
Hello again @mepox 
Sorry, I'm a little bit confused now. 
Firstly I made auto lines indentation (usually I'm doing it by pressing a hotkey in my IDE, it's a usual routine although I forgot to do it this time after deleting whole lines) both in HistoryController and ExerciseController.
Then I committed it:  [dd954e2](https://github.com/mepox/fitbuddy/pull/90/commits/dd954e2962596dac6e7248b767680e8f67b7d966)
And I saw that it changed the indentation interval sizes in whole source files altogether. 
So I reverted the commit and tried to auto-indent only highlighted methods: [f22ebf5](https://github.com/mepox/fitbuddy/pull/90/commits/f22ebf5eb392bb481c7621ffbd6fa708abe081ae)

I think I have to re-make it from the scratch. Could you please guide me how to better do it, do we have different indentation size settings or something else maybe?

**Comment by dmitriydb** [comment]
@mepox 

I've prepared another PR that does both things (removing auth null checks and indentation issues)
#99 

Could you please review it?

**Comment by mepox** [comment]
> Hello again @mepox Sorry, I'm a little bit confused now. Firstly I made auto lines indentation (usually I'm doing it by pressing a hotkey in my IDE, it's a usual routine although I forgot to do it this time after deleting whole lines) both in HistoryController and ExerciseController. Then I committed it: [dd954e2](https://github.com/mepox/fitbuddy/pull/90/commits/dd954e2962596dac6e7248b767680e8f67b7d966) And I saw that it changed the indentation interval sizes in whole source files altogether. So I reverted the commit and tried to auto-indent only highlighted methods: [f22ebf5](https://github.com/mepox/fitbuddy/pull/90/commits/f22ebf5eb392bb481c7621ffbd6fa708abe081ae)
> 
> I think I have to re-make it from the scratch. Could you please guide me how to better do it, do we have different indentation size settings or something else maybe?

Hi @dmitriydb ! You made a good job to try to fix the problem in this PR. Only one method identation wasn't good enough, so you were very close! However to avoid requesting changes again, I accepted your other PR as that was ready to merge.

Re your code formatting question, collaboration and IDE auto code formatters are not the best friends 😅 Why? Well, everyone has different code style/taste and they set up their IDE differently... Some people like identation size 2 others go with 4. Some people like to use tabs but others prefer to use spaces for tabs... 

Best advice is when contributing to a public project is to try and follow the project's code style so your code fits in nicely with the rest. This means either disable/not use the auto code formatter or configure the code formatter to match the project style.

Here we use indentation size 4 with tabs. Maybe I should add this to the docs 🤔 Looking at your code, you are probably using indentation size 2.

**Comment by dmitriydb** [comment]
@mepox 
Thank you very much for the detailed answer! It is very useful lesson

**Pull request closed without merging**